security.md

```diff
@@ -0,0 +1,94 @@
+"*password authenticated* key exchange", PAKE.)
+
+### Password entropy
+
+Compared to password strengths we are used to nowadays, 16 bits may seem scarily
+small, even knowing that an attacker only has one try to guess the password. But
+to put it into perspective: As a user, you'd have to experience 32000 failed
+transfers because of an attack in order to having a 50% chance of a file being
+intercepted. From an attacker's perspective, guessing thousands of codes is not
+difficult, but there is no possibility of targetting any individual connection,
+making any gained data fairly useless. From a service perspective, any successful
+attack implies 65536 failed attempts on average, which will quickly be noticed
+due to their disrupting nature.
+
+In short, guessing codes is unlikely to be an attractive target for any attacker.
+Of course, there is always the possibility to configure a longer code in the
+clients if desired.
+
+## Denial of Service (DoS) attacks on the Rendezvous server
+
+Wormhole codes can be so short because they implicitly contain a common
+rendezvous server URL (any two applications that use Magic Wormhole
+should be configured to use the same server). As a result, successful
+operation depends upon both clients being able to contact that server,
+making it a single point of failure (SPOF).
+
+While it is of course possible to incapacitate the rendezvous server using
+traditional means of DoS, so much work is not necessary: An attacker may
+simply connect to every nameplate with a random code, causing the key exchange
+to fail. There is a "list" command in the protocol which makes it easy to
+enumerate all nameplates to disrupt them, but even without it the name space is
+sufficiently small (by design, as we want short codes) to brute force.
+
+There are several ideas to defend against this, however the only one already
+specified in the protocol is the "permission" feature: The rendezvous server
+may use it to restrict new connections with some kind of challenge (or for
+private servers, authentication). The motivation behind this is to enable
+a proof-of-work challenge (using HashCash) when under attack to slow it down.
+See the server protocol specification for details.
+
+Currently the best defense against this is to use high-entropy nameplates which
+cannot easily be enumerated at the cost of longer codes. This has not be
+implemented yet though, and would still leave the attack vector of "ordinary"
+DoS.
+
+## Metadata leaks
+
+As usual in cryptography, communication metadata cannot easily be protected. Any
+involved server will know the timestamp and size of each message it relays. This
+usually is not much of an issue, except for file transfer where the file size may
+be characteristic for its content. (Imagine people using Magic Wormhole to send
+some specific movie.)
+
+Thanks to the separation between rendezvous server and relay server, forcing the
```

Comment:
Other places I believe we'd agreed that "mailbox" and "transit" are less-confusing names; perhaps we should use those here now too.